code/languages/C++/vector/README.md

```diff
@@ -0,0 +1,150 @@
+# Vector 
+We can think of **vectors** as smart arrays. They manage storage allocation for us, expanding
+and contracting the size of the vector as we insert or erase data. We can use vectors much like
+arrays, accessing elements with the [] operator. Such random access is very fast with vectors.
+It’s also fast to add (or push) a new data item onto the end (the back) of the vector. When this
+happens, the vector’s size is automatically increased to hold the new item. 
+
+## Member Functions push_back() , size() , and operator[]
+
+// demonstrates push_back(), operator[], size()
+
+```C++
+#include <iostream>
+#include <vector>
+
+int main()
+{
+  vector<int> v;          //create a vector of ints
+  //put values at end of array
+  v.push_back(10);
+  v.push_back(11);
+  v.push_back(12);
+  v.push_back(13);
+  //replace with new values 
+  v[0] = 20;
+  v[3] = 23; 
+
+  for(int j=0; j<v.size(); j++)     //display vector contents
+  {
+  std::cout << v[j] << ‘ ‘;        //20 11 12 23
+  std::cout << endl;
+  return 0;
```

Comment:
Format these too. Basically paste the code into some autoformatter and paste it back here.

Reply:
Done!